And indeed, my regression test [fails on Travis](https://travis-ci.org/numpy/numpy/jobs/383039061):
```
_____________ TestDateTime.test_timedelta_object_array_conversion ______________
self = <numpy.core.tests.test_datetime.TestDateTime object at 0x7f140fa8e1d0>
    def test_timedelta_object_array_conversion(self):
        # Regression test for GH11096
        inputs = [datetime.timedelta(28),
                  datetime.timedelta(30),
                  datetime.timedelta(31)]
        expected = np.array([28, 30, 31], dtype='timedelta64[D]')
>       actual = np.array(inputs, dtype='timedelta64[D]')
E       TypeError: Cannot cast datetime.timedelta object from metadata [Y] to [D] according to the rule 'same_kind'
```

--- a/numpy/core/tests/test_datetime.py
+++ b/numpy/core/tests/test_datetime.py
@@ -327,6 +327,12 @@ class TestDateTime(object):
         a = datetime.timedelta(seconds=3)
         assert_raises(TypeError, np.timedelta64, a, 'M')
         assert_raises(TypeError, np.timedelta64, a, 'Y')
+        a = datetime.timedelta(weeks=3)
+        assert_raises(TypeError, np.timedelta64, a, 'M')
+        assert_raises(TypeError, np.timedelta64, a, 'Y')
+        a = datetime.timedelta()
+        assert_raises(TypeError, np.timedelta64, a, 'M')
+        assert_raises(TypeError, np.timedelta64, a, 'Y')
 
     def test_timedelta_object_array_conversion(self):
         # Regression test for gh-11096

--- a/numpy/core/tests/test_datetime.py
+++ b/numpy/core/tests/test_datetime.py
@@ -324,9 +324,12 @@ class TestDateTime(object):
         a = np.timedelta64(1, 'Y')
         assert_raises(TypeError, np.timedelta64, a, 'D')
         assert_raises(TypeError, np.timedelta64, a, 'm')
+        a = datetime.timedelta(seconds=3)
+        assert_raises(TypeError, np.timedelta64, a, 'M')
+        assert_raises(TypeError, np.timedelta64, a, 'Y')
 
     def test_timedelta_object_array_conversion(self):
-        # Regression test for GH11096
+        # Regression test for gh-11096
         inputs = [datetime.timedelta(28),
                   datetime.timedelta(30),
                   datetime.timedelta(31)]

